### PR TITLE
Add Prow config for Hypershift ops repo

### PR DESCRIPTION
**Pull request by alvaroaleman**

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/hold
to wait for the rehearsal

File: ci-operator/config/openshift/hypershift-ops/openshift-hypershift-ops-main.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+base_images:
+  cli:
+    name: "4.11"
+    namespace: ocp
+    tag: cli
+build_root:
+  image_stream_tag:
+    as: this-is-unused-but-still-has-to-be-specified
```
Comment (ironcladlou) [inline ci-operator/config/openshift/hypershift-ops/openshift-hypershift-ops-main.yaml:8]: 😆

File: ci-operator/jobs/openshift/hypershift-ops/openshift-hypershift-ops-main-postsubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    always_run: true
+    branches:
+    - ^main$
+    - ^main-
+    cluster: build01
+    context: ci/prow/manifests-apply
+    decorate: true
+    labels:
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-hypershift-ops-main-manifests-apply-dry-run
```
Comment (ironcladlou) [inline ci-operator/jobs/openshift/hypershift-ops/openshift-hypershift-ops-main-postsubmits.yaml:13]: this is called "-dry-run" but it looks like it's doing the real apply

**Comment by ironcladlou** [comment]
/lgtm

feel free to remove hold when ready

**Pull request merged**